pkg/management/postgres/initdb.go

```diff
@@ -132,47 +132,85 @@ type InitInfo struct {
 //     important user data. This is particularly relevant when using static provisioning
 //     of PersistentVolumeClaims (PVCs), as it prevents accidental overwriting of a valid
 //     data directory that may exist in the PersistentVolumes (PVs).
-func (info InitInfo) CheckTargetDataDirectory(ctx context.Context) error {
+func (info InitInfo) CheckTargetDataDirectories(ctx context.Context) error {
 	contextLogger := log.FromContext(ctx).WithValues("pgdata", info.PgData)
 
 	pgDataExists, err := fileutils.FileExists(info.PgData)
 	if err != nil {
 		contextLogger.Error(err, "Error while checking for an existing PGData")
 		return fmt.Errorf("while verifying is PGDATA exists: %w", err)
 	}
-	if !pgDataExists {
-		// The PGDATA directory doesn't exist. We can definitely
-		// write to it
+
+	pgWalExists := false
+	if info.PgWal != "" {
+		if pgWalExists, err = fileutils.FileExists(info.PgWal); err != nil {
+			contextLogger.Error(err, "Error while checking for an existing PGWAL")
+			return fmt.Errorf("while verifying if PGWAL exists: %w", err)
+		}
+	}
+
+	if !pgDataExists && !pgWalExists {
 		return nil
 	}
 
-	// We've an existing directory. Let's check if this is a real
-	// PGDATA directory or not.
-	out, err := info.GetInstance().GetPgControldata()
-	if err != nil {
-		contextLogger.Info("pg_controldata check on existing directory failed, cleaning it up",
-			"out", out, "err", err)
+	if pgDataExists {
+		if out, err := info.GetInstance().GetPgControldata(); err != nil {
+			contextLogger.Info("pg_controldata check on existing directory failed, cleaning it up",
+				"out", out, "err", err)
 
-		if err := fileutils.RemoveDirectory(info.PgData); err != nil {
-			contextLogger.Error(err, "error while cleaning up existing data directory")
-			return err
+			if err := fileutils.RemoveDirectory(info.PgData); err != nil {
+				contextLogger.Error(err, "error while cleaning up existing data directory")
+				return err
+			}
+		} else {
+			contextLogger.Info("pg_controldata check on existing directory succeeded, renaming the folders")
+			return info.renameExistingTargetDataDirectories(ctx)
 		}
+	}
 
-		return nil
+	if pgWalExists {
+		contextLogger.Info("cleaning up existing PG_WAL directory")
+		if err := fileutils.RemoveDirectory(info.PgWal); err != nil {
+			contextLogger.Error(err, "error while cleaning up existing PG_WAL directory")
+			return err
+		}
```

Comment:
I find this control flow unclear. It's correct, but:
- if the PGDATA is there with a valid pg_control, we exit from this function into the "rename" function.
- if pg_control is NOT there, we RemoveDirectory of the PGDATA in the first conditional, and the possibl RemoveDirectory of the PGWAL in a different loop.

Going to try to propose a clearer flow.

Reply:
OK, I've refactored this control flow.
Please @mnencia have a look.